### PR TITLE
[CPDLP-1040] Stop payable declarations reverting to eligible

### PR DESCRIPTION
**Pull request by asmega**
## Ticket and context

Ticket: https://dfedigital.atlassian.net/browse/CPDLP-1040

- This fixes a bug where by a provider could submit a declaration for an existing payable declaration causing it to revert back to eligible
- All incoming declarations now create a new record regardless if another similar one exists or not
- We now create a DeclarationState record when a ParticipantDeclation is created as an ActiveRecord callback

### Code quality checks
- [x] All commit messages are meaningful and true
- [x] Added enough automated tests

### HTML Checks
- [x] All new pages have automated accessibility checks
- [x] All new pages have visual tests via Percy

### Gotchas
- [x] All `School` queries are correctly scoped - `eligible` when they need to be

### How can someone see it it review app?

- Find a declaration in payable or paid state
- Submit the same declaration again as the provider
- NOTE: timestamps MUST be identical otherwise it will be considered as a different declaration and you get some kind of error which doesn't test this edge case
- It will create another declaration that is ineligible
- The original declaration will remain in original state of payable or paid

## External API changes

- Nothing that needs documenting

### CODE REVIEW

File: app/models/participant_declaration.rb [hunk 1]
```diff
@@ -81,6 +81,8 @@ class ParticipantDeclaration < ApplicationRecord
   scope :paid_npqs_for_lead_provider, ->(lead_provider) { paid_for_lead_provider(lead_provider).npq }
   scope :paid_uplift_for_lead_provider, ->(lead_provider) { paid_for_lead_provider(lead_provider).uplift }
 
+  after_create :create_initial_declaration_state
```
Comment (jamesrose) [inline app/models/participant_declaration.rb:84]: Do you want this build as part of the create? (e.g., `before_create` and `declaration_states.build(...)`)

Comment (asmega) [reply app/models/participant_declaration.rb:84]: - i'm not sure `before_create` would work as it needs to reference the persisted record.
- `build` i think could work but where would you stick it? and also what would you set the state to when you call `build` as it could be `submitted` or `ineligible`

Comment (jamesrose) [reply app/models/participant_declaration.rb:84]:
```
  before_create :build_initial_declaration_state
  
  private
  
  def build_initial_declaration_state
    declaration_states.build(state: state)
  end
```

I _think_ this should work if `inverse_of` is correctly set on each side of the association. The save on the declaration will save the state.

Comment (asmega) [reply app/models/participant_declaration.rb]: Worked without `inverse_of `. Saving the parent seems to also save the child

**Review by jamesrose — APPROVED** [review]
